### DISCUSSION TITLE
Photoprism Reboots Host Device

### DISCUSSION THREAD

**Discussion by ashaychangwani** [photoprism/photoprism#3236] [Q&A]
I have Photoprism and Mariadb installed via docker-compose, right from the user guide on my Raspberry Pi 4 (4gb). I have set the swap space to 2gb since the guide recommends that.

The library currently has 0 photos.

However, as soon as I upload something to photoprism, my entire raspberry pi reboots even though photoprism is just running in a docker container. 

I have already tried everything in the troubleshooting steps and the logs also don't show anything suspicious. How can I debug this?

**Comment by Anderen2** [comment]
Sounds like a power delivery issue, are you sure that the PSU powering your Raspberry is up to spec and working correctly? Try connecting up a different power supply to check. 
There should be no way for Photoprism to reboot the host.

**Comment by williamwd** [comment]
I was having the same issue with a full-size Dell server. 
Is there a reason for the `docker-compose.yml` to define the `security_opt` section? There really should be no way of the container restarting the host, but running `[unconfined](seccomp:unconfined)` invalidates this statement.

**Reply by lastzero** [reply]
Most of the time there's some other underlying problem if this happens. This setting would generally not lead to a restart and it's the first time I hear about this. It part of our default configuration since we provide our software for free to home users and so can't provide individual help with configuring kernel security modules along with it. Our docs also state that experienced users may not need this.

**Reply by williamwd** [reply]
Just the removal of the `security_opt` section from `docker-compose.yml` solved the issue for me. Observed the last couple of days, no reboots.

Didn't have an opportunity to look into the implications of doing so, though. I'm sure you put that section for a reason, that's why I asked.

I know that you provide this fantastic software for free, and I truly appreciate it. All I'm saying is that there I could confirm the behaviour was being caused by Photoprism, since I had just started using it a couple of months ago and have never experienced issues with my server. 
It also happened on my desktop when I first tested. But, I didn't think much of it, since it needed some cleaning to stop overheating.

**Reply by lastzero** [reply]
It MAY be needed (or WAS needed) in combination with certain Docker versions and/or Linux distributions and third-party software that PhotoPrism MAY use, such as Darktable or MariaDB. Unfortunately our team is too small to test it with every conceivable combination of Docker, Linux distribution, and third-party software, which is why that's part of our default configuration. Note that PhotoPrism itself does not need it and a properly configured Docker installation very likely does not need it as well. Should it now be that this option causes MORE instead of LESS unpaid support overhead for us, we will be happy to remove it.
